File: 剑指offer代码自编/14_reorderOddEven.py
def reorderOddEven(a):
	low = 0;
	high = len(a) - 1
	while(low < high):
		while(low < high and a[low] % 2 != 0):
			low += 1
		while(low < high and a[high] % 2 == 0):
			high -= 1
		if(low < high):
			temp = a[low]
			a[low] = a[high]
			a[high] = temp
	return a
print reorderOddEven([1,3,2,4])

File: 剑指offer代码/14_reorderOddEven.cpp
void reOrderOddEven(int* data, int length)
{
	if(data == NULL || length == 0)
		return;
	int* begin = data;
	int* end = data + length - 1;
	while(begin < end)
	{
		while(*begin % 2 != 0)
			begin ++;
		while(*end %2 == 0)
			end --;
		if(begin < end)
		{
			int temp = *begin;
			*begin = *end;
			*end = temp;
		}
	}
}